Pipelines/Pipeline.yml

```diff
@@ -0,0 +1,28 @@
+name: CD-ApiMDeveloperPortal-Build
+
+trigger: none
+
+variables:
+  - name: poolName
+    value: "agentpoolName"
+  - name: location
+    value: West US
+
+jobs:
+
+  # All tasks on APIM Developer portal pipeline
+  - job: Deploy_APIM_Developer_Portal
+    displayName: Deploy APIM Developer portal from one APIM instance to another APIM instance Migration 
+    pool:
+      name: $(poolName)
+    timeoutInMinutes: 90
+    steps:
+      - task: Npm@1
+        displayName: Npm Install command
+        inputs:
+          command: "install"
+
+      - pwsh: |        
+          node ./migrate --sourceSubscriptionId "$(sourceSubscriptionId)" --sourceResourceGroupName  "$(sourceResourceGroupName)" --sourceServiceName "$(sourceAPIMName)" --destServiceName "$(destinationAPIMName)" --destSubscriptionId "$(destSubscriptionId)" --destResourceGroupName "$(destResourceGroupName)" --sourceTenantid "$(sourceAzure_Tenant)" --sourceServiceprincipal "$(sourceServicePrincipal)" --sourceSecret "$(sourceAzureDevOps-ServicePrincipal-Secret)"  --destTenantid "$(Azure_Tenant)" --destServiceprincipal "$(ServicePrincipal)" --destSecret "$(AzureDevOps-ServicePrincipal-Secret)" --existingEnvUrls "$(existingEnvUrls)" --destEnvUrls  "$(destEnvUrls)"
```

Comment:
I'm not expert, but I think these variable should be declared in variables section, like [documentation](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#macro-syntax-variables) suggests. Example:
```
variables:
 - name: projectName
   value: contoso

steps: 
- bash: echo $(projectName)
- powershell: echo $(projectName)
- script: echo $(projectName)
```

Reply:
Added readme file with details of pipeline variables to be added with value descriptions.